data/uistrings.yaml

```diff
@@ -0,0 +1,143 @@
+  services, like those for debugging or development, to be included in a single
+  `compose.yml` file and activated only as needed.
+composeSecrets:
+  Docker Compose provides a way for you to use secrets without having to use
+  environment variables to store information. If you’re injecting passwords and
+  API keys as environment variables, you risk unintentional information
+  exposure. Services can only access secrets when explicitly granted by a
+  `secrets` attribute within the `services` top-level element.
+composeServices:
+  A service is an abstract definition of a computing resource within an
+  application which can be scaled or replaced independently from other
+  components. Services are backed by a set of containers, run by the platform
+  according to replication requirements and placement constraints. As services
+  are backed by containers, they are defined by a Docker image and set of
+  runtime arguments. All containers within a service are identically created
+  with these arguments.
+composeServicesDependsOn:
+  With the `depends_on` attribute, you can control the order of service startup
+  and shutdown. It is useful if services are closely coupled, and the startup
+  sequence impacts the application's functionality.
+composeServicesDevelop:
+  Develop lets Compose focus on the development use-case of running
+  applications on a local machine. It also supports some development hooks to
+  improve the velocity of your local workflow, also known as your "inner loop". 
+composeServicesEnvFile:
+  The `env_file` attribute is used to specify one or more files that contain
+  environment variables to be passed to the containers.
+composeServicesEnvironment:
+  The `environment` attribute defines environment variables set in the
+  container. `environment` can use either an array or a map. Any boolean
+  values; true, false, yes, no, should be enclosed in quotes to ensure they are
+  not converted to True or False by the YAML parser.
+composeServicesHealthcheck:
+  The `healthcheck` attribute declares a check that's run to determine whether
+  or not the service containers are "healthy". It works in the same way, and
+  has the same default values, as the [HEALTHCHECK Dockerfile
+  instruction](https://docs.docker.com/reference/dockerfile/#healthcheck) set
+  by the service's Docker image. Your Compose file can override the values set
+  in the Dockerfile. 
+composeServicesNetworks:
+  The `networks` attribute defines the networks that service containers are
+  attached to, referencing entries under the [`networks` top-level
+  element](06-networks.md). The `networks` attribute helps manage the
+  networking aspects of containers, providing control over how services are
+  segmented and interact within the Docker environment. This is used to specify
+  which networks the containers for that service should connect to. This is
+  important for defining how containers communicate with each other and
+  externally.
```

Comment:
Honestly, slightly on the fence if it's great to have to write Markdown in YAML 🤔 

Admitted, the current `includes/` are really short snippets, so maybe not perfect either though. Could we generate the YAML file? Would that make it more flexible? (Also considering if we could even have some markup in pages that would allow extracting snippets, e.g. some `<! --- snippet:compose.hello-world -->` delimiter that would allow extracting sections that can be reused.

I know we also looked for the Dockerfile syntax-links (i.e.. the links on `FROM` etc) to show some pop-up; see   https://github.com/docker/docs/pull/13369#issuecomment-900594509 and https://github.com/docker/docs/pull/13485)

Reply:
We don't really need to yaml per se. It's just a way to assign UI strings a unique ID (e.g. `composeServicesNetworks`) that applications can reference and that won't break if we move a file.

Defining reusable snippets inline might be dangerous because this can potentially make it difficult to track down the source of a snippet, and also lead to duplicates. (And there's no standard way to do this in Hugo, so we'd end up with a regex hack to extract them.)

```html
<!-- snippet:compose.hello-world -->
```

I think it depends on what the content is. If it's short strings, I think managing them as YAML makes sense. If it's more, or if it should include things like images (or links, as we can already see), then a YAML multiline string doesn't feel like the perfect solution.

Reply:
Part of me thinks that we should just stand up an actual headless CMS for this.